src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,30 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+
+type Props = {
+  userTodo: Todo | null;
+  resetSelection: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ userTodo, resetSelection }) => {
+  const [isLoaderShowing, setIsLoaderShowing] = useState(true);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(userTodo?.userId || 0).then(user => {
+      setSelectedUser(user);
+      setIsLoaderShowing(false);
+    });
```

Comment:
```suggestion
  const [isLoaderShowing, setIsLoaderShowing] = useState(false);
  const [selectedUser, setSelectedUser] = useState<User | null>(null);

  useEffect(() => {
    setIsLoaderShowing(true);
    
    getUser(userTodo?.userId || 0)
      .then(setSelectedUser)
      .catch(error => // handler errors)
      .finally(() => {
        setIsLoaderShowing(false);
      });
```
You should show the loader not by default, but only in case you actually loading something
And as I mentioned before you should show the loader when you loading tods as well